modules/nixos/main.py

```diff
@@ -23,316 +23,44 @@
-
-cfgunfree = """  # Allow unfree packages
-  nixpkgs.config.allowUnfree = true;
-
-"""
-
-cfgpkgs = """  # List packages installed in system profile. To search, run:
-  # $ nix search wget
-  environment.systemPackages = with pkgs; [
-  #  vim # Do not forget to add an editor to edit configuration.nix! The Nano editor is also installed by default.
-  #  wget
-  ];
-
-"""
-
-cfgtail = """  # Some programs need SUID wrappers, can be configured further or are
-  # started in user sessions.
-  # programs.mtr.enable = true;
-  # programs.gnupg.agent = {
-  #   enable = true;
-  #   enableSSHSupport = true;
-  # };
-
-  # List services that you want to enable:
-
-  # Enable the OpenSSH daemon.
-  # services.openssh.enable = true;
-
-  # Open ports in the firewall.
-  # networking.firewall.allowedTCPPorts = [ ... ];
-  # networking.firewall.allowedUDPPorts = [ ... ];
-  # Or disable the firewall altogether.
-  # networking.firewall.enable = false;
-
-  # This value determines the NixOS release from which the default
-  # settings for stateful data, like file locations and database versions
-  # on your system were taken. It‘s perfectly fine and recommended to leave
-  # this value at the release version of the first install of this system.
-  # Before changing this value read the documentation for this option
-  # (e.g. man configuration.nix or on https://nixos.org/nixos/options.html).
-  system.stateVersion = "@@nixosversion@@"; # Did you read the comment?
-
-}
-"""
-
-cfglatestkernel = """  # Use latest kernel.
-  boot.kernelPackages = pkgs.linuxPackages_latest;
+    if os.path.isfile(customConfigFilePath):
+        libcalamares.utils.debug('Selected custom snippet: {} from {}'.format(snippetName, customConfigFilePath))
+        snippetConfigPath = customConfigFilePath
+    else:
+        libcalamares.utils.debug('Selected default snippet: {} from {}'.format(snippetName, snippetConfigPath))
+
+    # load config
+    with open(snippetConfigPath, 'r') as configReader:
+        snippet = configReader.read()
+        snippet += '\n'
+        return snippet
+
+    libcalamares.utils.error('Failed to load config snippet {}'.format(snippetName))
+    return ""
+
+def copyModules(root_mount_point, subpath):
+    srcModuleFolderPath = '{}/{}/.'.format(getCalamaresNixosModulePath(), subpath)
+    destModuleFolderPath = '{}/etc/nixos/modules/'.format(root_mount_point)
+    if os.path.isdir(srcModuleFolderPath) and len(os.listdir(srcModuleFolderPath)) != 0:
+        if not os.path.isdir(destModuleFolderPath):
+            libcalamares.utils.host_env_process_output(
+                ["mkdir", "-p", destModuleFolderPath], None
+            )
+        libcalamares.utils.debug('Copying module directory {}'.format(srcModuleFolderPath))
+        libcalamares.utils.host_env_process_output(["cp", "-r", srcModuleFolderPath, destModuleFolderPath], None)
+    else:
+        libcalamares.utils.debug('Skipping non-existing module directory {}'.format(srcModuleFolderPath))
```

Comment:
I don't know why but this still seems to be creating an empty `/etc/nixos/modules` directory.